firmware/application/apps/capture_app.cpp

```diff
@@ -51,28 +51,27 @@ CaptureAppView::CaptureAppView(NavigationView& nav)
     field_frequency_step.set_by_value(receiver_model.frequency_step());
     field_frequency_step.on_change = [this](size_t, OptionsField::value_t v) {
         receiver_model.set_frequency_step(v);
-        this->field_frequency.set_step(v);
     };
 
     option_bandwidth.on_change = [this](size_t, uint32_t base_rate) {
-        sampling_rate = 8 * base_rate;  // Decimation by 8 done on baseband side
-                                        /* base_rate is used for FFT calculation and display LCD, and also in recording writing SD Card rate. */
+        /* base_rate is used for FFT calculation and display LCD, and also in recording writing SD Card rate. */
         /* ex. sampling_rate values, 4Mhz, when recording 500 kHz (BW) and fs 8 Mhz, when selected 1 Mhz BW ... */
         /* ex. recording 500kHz BW to .C16 file, base_rate clock 500kHz x2(I,Q) x 2 bytes (int signed) =2MB/sec rate SD Card  */
-        waterfall.on_hide();
+        auto sampling_rate = 8 * base_rate;  // Decimation by 8 done on baseband side.
 
         /* Set up proper anti aliasing BPF bandwith in MAX2837 before ADC sampling according to the new added BW Options. */
         auto anti_alias_baseband_bandwidth_filter = filter_bandwidth_for_sampling_rate(sampling_rate);
 
+        baseband::spectrum_streaming_stop();
         record_view.set_sampling_rate(sampling_rate);
         receiver_model.set_sampling_rate(sampling_rate);
         receiver_model.set_baseband_bandwidth(anti_alias_baseband_bandwidth_filter);
-
-        waterfall.on_show();
+        // TODO: baseband::set_spectrum too?
+        baseband::spectrum_streaming_start();
     };
 
-    option_bandwidth.set_selected_index(7);  // Preselected default option 500kHz.
     receiver_model.enable();
```

Comment:
These changes didn't resolve the problem for me either, but inserting chThdSleepMilliseconds(1) before the call to receiver_model.enable() seems to do the trick for some reason.

Reply:
[Hope you don't mind but I marked this PR as a Draft since it doesn't work yet]